src/gausskernel/storage/access/k2/storage.cpp

```diff
@@ -121,19 +79,15 @@ class UntoastedDatum {
     }
 };
 
-K2PgStatus getSKVBuilder(K2PgOid database_oid, K2PgOid table_oid, std::shared_ptr<k2pg::catalog::SqlCatalogClient> catalog,
+K2PgStatus getSKVBuilder(K2PgOid database_oid, K2PgOid table_oid,
                          std::unique_ptr<skv::http::dto::SKVRecordBuilder>& builder) {
-    std::string collectionName;
-    std::string schemaName;
-
-    skv::http::Status catalog_status = catalog->GetCollectionNameAndSchemaName(database_oid, table_oid, collectionName, schemaName);
-    if (!catalog_status.is2xxOK()) {
-        return K2StatusToK2PgStatus(std::move(catalog_status));
-    }
+    std::shared_ptr<k2pg::PgTableDesc> pg_table = k2pg::pg_session->LoadTable(database_oid, table_oid);
+    std::string collectionName = std::move(pg_table->collection_name());
+    std::string schemaName = std::move(pg_table->schema_name());
```

Comment:
I don't think it is safe to move these since pg_table is shared through the shared_ptr

Reply:
changed to use const std::string& instead of using std::move()